src/classes/CMT_MetadataAPI.cls

```diff
@@ -93,39 +93,60 @@ public class CMT_MetadataAPI {
         });
     }
 
-    /**
+    /******************************************************************************************************************
      * @description Initiate a deployment of custom metadata type records.
      * @param jobId Unique string to act as a "jobId" when writing the final deployment status to the Custom Settings
      * @param mtToDeploy List of CustomMetadata objects to deploy
-     */
-    public static void deployMetadata(String jobId, List<Metadata.CustomMetadata> mtToDeploy) {
+     * @param callback a MetadataCallBack to handle the deployment result
+     *****************************************************************************************************************/
+    public static void deployMetadata(String jobId, List<Metadata.CustomMetadata> mtToDeploy, MetadataCallBack callback) {
 
         // Add all queued up metadata updates to the deployment container
         Metadata.DeployContainer mdContainer = new Metadata.DeployContainer();
         for (Metadata.CustomMetadata md: mtToDeploy) {
-            system.debug('deployMetadata > ' + md.fullName);
             mdContainer.addMetadata(md);
         }
 
-        // Setup deploy callback
-        MetadataCallBack callback = new MetadataCallBack(jobId);
-
         // Enqueue custom metadata deployment and return the Job Id
         if (!Test.isRunningTest()) {
             Metadata.Operations.enqueueDeployment(mdContainer, callback);
         }
     }
 
-    /**
+    /******************************************************************************************************************
+     * @description Initiate a deployment of custom metadata type records.
+     * @param jobId Unique string to act as a "jobId" when writing the final deployment status to the Custom Settings
+     * @param mtToDeploy List of CustomMetadata objects to deploy
+     *****************************************************************************************************************/
+      public static void deployMetadata(String jobId, List<Metadata.CustomMetadata> mtToDeploy) {
+          deployMetadata(jobId, mtToDeploy, new MetadataCallBack(jobId));
+      }
+
+
+    /******************************************************************************************************************
+     * @description Interface for Metadata Callback Handler
+     * Is used to allow metadata deployments to have customized on result handlers
+     *****************************************************************************************************************/
+    public interface MetadataCallbackHandler {
+        void performSuccessHandler(Map<String, Object> params);
+    }
+
+    /******************************************************************************************************************
```

Comment:
In order to execute the invalidation cache after a successful CMT deployment - I created an interface and added it to the Custom Callback class we already created.  Right now it only handles on success but we can extend it to provide on error handler as well.